sync-core/src/main/java/com/cloudant/mazha/CouchClient.java

```diff
@@ -61,29 +58,72 @@ public URI getRootUri() {
         return this.uriHelper.getRootUri();
     }
 
-    HttpRequests getHttpClient() {
-        return httpClient;
-    }
+    // - if 2xx then return stream
+    // - map 404 to NoResourceException
+    // - if there's a couch error returned as json, unmarshall and throw
+    // - anything else, just throw the IOException back, use the cause part of the exception?
 
-    public List<String> getList(URI uri) {
-        InputStream is = null;
+    // it needs to catch eg FileNotFoundException and rethrow to emulate the previous exception handling behaviour
+    private InputStream executeToInputStream(HttpConnection connection) throws CouchException {
+
+        // all couchclient requests want to receive application/json responses
+        connection.requestProperties.put("Accept", "application/json");
```

Comment:
I think all the `GET` etc. requests already set this where needed, or they probably should do?